plugins/Carbonix/UI/TakeoffTab.cs

```diff
@@ -443,6 +451,7 @@ private void but_safety_Click(object sender, EventArgs e)
             ((Control)sender).Enabled = false;
 
             set_safety_switch(!Host.cs.safetyactive);
+            SendEngineCommand(false);

```

Comment:
```suggestion
            // When we engage the safety, send an engine kill command so the engine doesn't try to restart as soon as we disengage safety
            if(!Host.cs.safetyactive)
            {
                SendEngineCommand(false);
            }
            set_safety_switch(!Host.cs.safetyactive);
```

Reply:
I had it like this for the case where if you had tried to crank the engine with safety on and then turned off the safety it wouldn't immediately crank. I can just change the order but I think there is some value of sending a kill command on all toggles of the safety.

Reply:
I added a change text on the button and change the running state